yt_dlp/extractor/vk.py

```diff
@@ -606,39 +640,109 @@ class VKUserVideosIE(VKBaseIE):
-        video_list = video_list_json['list']
+    API_BASE = 'https://api.vk.com/method'
+    API_Version = 5.269
+
+    def _get_playlist_payload_and_url(self, owner_id, album_id, access_token, offset=0):
+        url = f'{self.API_BASE}/video.getFromAlbum'
+        return url, {
+            'v': self.API_Version,
+            'album_id': album_id,
+            'fields': 'is_esia_verified,is_sber_verified,is_tinkoff_verified,photo_50,verified',
+            'owner_id': owner_id,
+            'access_token': access_token,
+            'offset': offset,
+            'count': 50,
+        }
+
+    def _get_user_payload_and_url(self, url, access_token, start_from=None, section_id=None):
+        if url:
+            return f'{self.API_BASE}/catalog.getVideo', {
+                'v': self.API_Version,
+                'need_blocks': 1,
+                'url': url,
+                'owner_id': 0,
+                'access_token': access_token,
+            }
+        else:
+            return f'{self.API_BASE}/catalog.getSection', {
+                'v': self.API_Version,
+                'need_blocks': 1,
+                'section_id': section_id,
+                'start_from': start_from,
+                'owner_id': 1,
+                'access_token': access_token,
+            }
 
+    def _parse_videos(self, data, video_id):
+        video_data = traverse_obj(data, ('files'), ('video', 'files'))
+        formats, subtitles = self._parse_formats_and_subtiles(video_id, video_data)
+        return {
+            'id': video_id,
+            'formats': formats,
+            'subtitles': subtitles,
+            'title': data.get('title'),
+            'description': data.get('description'),
+            'uploaded_date': data.get('date'),
+            'duration': data.get('duration'),
+            'comments': data.get('comments'),
+            'thumbnails': data.get('image'),
+            'views': data.get('views'),
+        }
+
+    def _entries(self, url, page_id, section, album_id=None):
+        access_token = self._get_access_token(page_id)
+        if section.startswith('play'):
+            url, payload = self._get_playlist_payload_and_url(page_id, album_id, access_token)
+        else:
+            url, payload = self._get_user_payload_and_url(url, access_token)
+
+        next_token = None
+        section_id = None
+        data = self._download_json(
+            url,
+            page_id,
+            query=payload,
+            headers={
+                'content-type': 'application/x-www-form-urlencoded',
+                'origin': 'https://vkvideo.ru',
+            },
+        )['response']
+        video_list = traverse_obj(data, ('catalog', 'videos', {list}), ('videos', {list}), ('items', {list}))
+        count = len(video_list)
+        total = traverse_obj(data, ('albums', 0, 'count', {int}), ('count', {int}))
+        next_token = traverse_obj(data, ('catalogsections', 0, 'next_from'), (('catalog', 'sections', 1, 'next_from')))
```

Comment:
```suggestion
        next_token = traverse_obj(data, ('catalogsections', 0, 'next_from'), ('catalog', 'sections', 1, 'next_from'))
```
supposing that `catalogsections` is correct?